raft/src/test/java/org/apache/kafka/raft/QuorumStateTest.java

```diff
@@ -45,7 +44,7 @@
     private final MockTime time = new MockTime();
     private final int electionTimeoutMs = 5000;
     private final int fetchTimeoutMs = 10000;
-    private final Random random = Mockito.spy(new Random(1));
+    private final MockableRandom random = new MockableRandom(1L);
```

Comment:
`spy` on `java.util.Random` doesn't work on Java 17 unless one opens an internal jdk package. I will file a `mockito` issue to understand if this is expected or if they can find a way around it. For these tests, it was easy enough to avoid the issue though.